src/main/java/com/hivemq/extensions/aws/cloudwatch/CloudWatchReporterService.java

```diff
@@ -63,21 +65,38 @@ void startCloudWatchReporter(
         } else {
             final Duration apiTimeout = cloudWatchConfig.getApiTimeout().map(Duration::ofMillis).orElse(null);
 
-            final CloudWatchAsyncClient cloudWatchAsync = CloudWatchAsyncClient.builder()
-                    .credentialsProvider(DefaultCredentialsProvider.create())
-                    .asyncConfiguration(ClientAsyncConfiguration.builder()
-                            .advancedOption(SdkAdvancedAsyncClientOption.FUTURE_COMPLETION_EXECUTOR, executorService)
-                            .build())
-                    .overrideConfiguration(ClientOverrideConfiguration.builder()
-                            .apiCallTimeout(apiTimeout)
-                            .apiCallAttemptTimeout(apiTimeout)
-                            .build())
-                    .build();
-
-            cloudWatchReporter = CloudWatchReporter.forRegistry(metricRegistry, cloudWatchAsync, METRIC_NAMESPACE)
+            final CloudWatchAsyncClientBuilder cloudWatchAsyncClientBuilder = CloudWatchAsyncClient.builder();
+            if (configuration.getConfig().getAwsEndpointOverride() != null) {
+                cloudWatchAsyncClientBuilder.endpointOverride(URI.create(configuration.getConfig()
+                        .getAwsEndpointOverride()));
+            }
+
+            final CloudWatchAsyncClient cloudWatchAsync =
+                    cloudWatchAsyncClientBuilder.credentialsProvider(DefaultCredentialsProvider.create())
+                            .asyncConfiguration(ClientAsyncConfiguration.builder()
+                                    .advancedOption(SdkAdvancedAsyncClientOption.FUTURE_COMPLETION_EXECUTOR,
+                                            executorService)
+                                    .build())
+                            .overrideConfiguration(ClientOverrideConfiguration.builder()
+                                    .apiCallTimeout(apiTimeout)
+                                    .apiCallAttemptTimeout(apiTimeout)
+                                    .build())
+                            .build();
+
+            final CloudWatchReporter.Builder builder =
+                    CloudWatchReporter.forRegistry(metricRegistry, cloudWatchAsync, METRIC_NAMESPACE);
+
+            if (configuration.getConfig().getZeroValuesSubmission()) {
+                builder.withZeroValuesSubmission();
+            }
+            if (configuration.getConfig().getReportRawCountValue()) {
+                builder.withReportRawCountValue();
+            }
+            cloudWatchReporter = builder.withZeroValuesSubmission()
+                    .withReportRawCountValue()
```

Comment:
```suggestion
            cloudWatchReporter = builder
```
you add them conditionally at the top, right?